backends/dnf/pk-backend-dnf.c

```diff
@@ -318,6 +335,10 @@ pk_backend_initialize (GKeyFile *conf, PkBackend *backend)
 						  g_free,
 						  (GDestroyNotify) dnf_sack_cache_item_free);
 
+	priv->sack_expire_source = g_timeout_add_seconds (SACK_TTL / 2,
```

Comment:
/2?

Reply:
The timeout and the sack cache expiration aren't necessarily going to align well, so if the check is done at the same interval as the TTL, then some sacks will probably live close to 2x the TTL.  Checking at TTL/2 means that the real max age is somewhere around 1.5x TTL.  Smaller intervals mean more frequent wakeups, so I tend to prefer this level of accuracy.